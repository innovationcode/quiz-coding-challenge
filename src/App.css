.App {
  text-align: center;
  background-image: linear-gradient(to top, #3c7baf 0%, #2be2be 100%);
  background-position: center; /* Center the image */
  background-repeat: no-repeat; /* Do not repeat the image */
  background-size: cover; /* Resize the background image to cover the entire container */
}

.data-display {
  display: grid;
  place-items: center;
  height: 80vh;
}

.data-display h3 {
  border: 1px solid green;
  width: 50vw;
  border-radius: 5px;
  padding: 5px;
  padding-left: 8px;
  box-shadow: 0 0 5px #999;
  display: flex;
  font-family: "Roboto";
  font-weight: bold;
  background-color: antiquewhite;
}
