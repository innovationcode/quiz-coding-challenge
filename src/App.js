import React from 'react';
import Header from './components/header/Header.js'
import QuestionDisplay from './components/questionDisplay/QuestionDisplay.js';
import { data } from './questionData/questionData.js'
import QuizResult from './components/quizResult/QuizResult.js';
import QuestionAndAnswer from './components/questionAndAnswer/QuestionAndAnswer.js';
import './App.css';


class App extends React.Component {
  state = {
      userAnswer: null,
      score: 0,
      currentIndex: 0,
      ansOptions: [],
      questionEnd: false,
      quizEnd: false
  }

  componentDidMount() {
    const { currentIndex } = this.state;
    this.setState(() => {
      return {
        question: data[currentIndex].question,
        ansOptions: [
          ...data[currentIndex].incorrect,
          data[currentIndex].correct
        ].sort(),
        correctAnswer: data[currentIndex].correct
      };
    });
  }

  nextQuestionHandler = () => {
    const { userAnswer, correctAnswer } = this.state;
    //console.log("userAnswer  : ", userAnswer ,"\ncorrectAnswer : ", correctAnswer)
    if (userAnswer === correctAnswer) {
      this.setState({
        score : this.state.score + 1 
      }, () => console.log("score in setState..  : - ",this.state.score)
      );
    }
    //console.log("score  : - ",this.state.score)
    this.setState({
      currentIndex: this.state.currentIndex + 1,
      userAnswer: null
    });
  };

  componentDidUpdate(prevProps, prevState) {
    const { currentIndex } = this.state;
    if (this.state.currentIndex !== prevState.currentIndex) {
      this.setState(() => {
        return {
          question: data[currentIndex].question,
          ansOptions: [
            ...data[currentIndex].incorrect,
            data[currentIndex].correct
          ].sort(),
          correctAnswer: data[currentIndex].correct
        };
      });
    }
  }

  checkAnswer = (answer) => {
    this.setState({
        userAnswer: answer,
    })
  }

  finishHandler = () => {
    const { userAnswer, correctAnswer } = this.state;
    //console.log("userAnswer  : ", userAnswer ,"\ncorrectAnswer : ", correctAnswer)
    if (userAnswer === correctAnswer) {
      this.setState({
        score : this.state.score + 1 
      }, () => console.log("score in setState..  : - ",this.state.score)
      );
    }

    if (this.state.currentIndex === data.length - 1) {
      this.setState({
        quizEnd: true
      });
    }
  };
  
  render() {
    const length = data.length;
    return (
      <div className="App"> 
        <Header />
 
        {!this.state.quizEnd ? (  
            <QuestionDisplay question = {this.state.question}
                ansOptions = {this.state.ansOptions}
                correctAnswer = {this.state.correctAnswer}
                currentIndex = {this.state.currentIndex}
                length= {length}
                nextQuestionHandler = {this.nextQuestionHandler}
                finishHandler = {this.finishHandler}
                checkAnswer ={this.checkAnswer}
                userAnswer = {this.state.userAnswer}
            />
          
        ) : (
          <>
            <QuizResult
                score = {this.state.score}
            />
            <QuestionAndAnswer /> 
          </>
        )}     
      </div>
    );
  }
}

export default App;
