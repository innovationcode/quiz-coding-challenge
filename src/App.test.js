import App from './App';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

//Two test written for App component

describe('tests App component.. ',() => {

    it("renders the correct content", () => {
        const root = document.createElement("div");
        ReactDOM.render(<App />, root);
    });

    it('matches the snapshot', () => {
        const tree = renderer.create(<App />).toJSON();
        expect(tree).toMatchSnapshot();
    })

});

    