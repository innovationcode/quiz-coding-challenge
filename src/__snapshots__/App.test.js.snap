// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests App component..  matches the snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="header"
  >
    Quiz Coding Challenge.
  </h1>
  <div
    className="questionDisplay"
  >
    <span
      style={
        Object {
          "fontSize": "16px",
          "fontWeight": "400",
        }
      }
    >
      Question 1 of 21
    </span>
    <h3
      className="question"
    >
      What was Tandem previous name?
    </h3>
    <p
      style={
        Object {
          "paddingBottom": "6px",
          "paddingTop": "12px",
        }
      }
    >
      The options are ...
    </p>
    <button
      className="ans-option-button null"
      onClick={[Function]}
    >
      Burger Shack
    </button>
    <button
      className="ans-option-button null"
      onClick={[Function]}
    >
      Devmynd
    </button>
    <button
      className="ans-option-button null"
      onClick={[Function]}
    >
      Extraordinary Humans
    </button>
    <button
      className="ans-option-button null"
      onClick={[Function]}
    >
      Tandem
    </button>
    <button
      className="next-button"
      onClick={[Function]}
    >
      Next
    </button>
  </div>
</div>
`;
